src/containers/Latest.js

```diff
@@ -0,0 +1,348 @@
+import Helmet from 'react-helmet'
+import PropTypes from 'prop-types'
+import querystring from 'querystring'
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+// components
+import SystemError from '../components/SystemError'
+import Pagination from '../components/Pagination'
+// utils
+import loggerFactory from '../logger'
+import { shallowCloneMetaOfPost } from '../utils/shallow-clone-entity'
+// constants
+import dataLoaderConst from '../constants/data-loaders'
+import siteMeta from '../constants/site-meta'
+// @twreporter
+import twreporterRedux from '@twreporter/redux'
+import mq from '@twreporter/core/lib/utils/media-query'
+import { List } from '@twreporter/react-components/lib/listing-page'
+import { TitleTab } from '@twreporter/react-components/lib/title-bar'
+// lodash
+import forEach from 'lodash/forEach'
+import get from 'lodash/get'
+import map from 'lodash/map'
+import findIndex from 'lodash/findIndex'
+import merge from 'lodash/merge'
+import concat from 'lodash/concat'
+const _ = {
+  forEach,
+  get,
+  map,
+  findIndex,
+  merge,
+  concat,
+}
+// global var
+const { actions, reduxStateFields } = twreporterRedux
+const { fetchPostsByTagListId, fetchLatestPosts, fetchLatestTags } = actions
+const logger = loggerFactory.getLogger()
+
+const Container = styled.div`
+  margin: 0 auto;
+  max-width: 1130px; //hd
+
+  ${mq.desktopOnly`
+    max-width: 922px;
+  `}
+  ${mq.tabletOnly`
+    max-width: 698px;
+  `}
+  ${mq.mobileOnly`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    max-width: 767px;
+  `}
+`
+
+const TitleTabContainer = styled.div`
+  margin: 64px 0;
+
+  ${mq.tabletOnly`
+    margin: 32px 0;
+  `}
+  ${mq.mobileOnly`
+    margin: 24px 0;
+    width: 86.66666666666667%;
+  `}
+`
+
+const Latest = ({
+  latestTagList,
+  activeTabIndex,
+  tagId,
+  pathname,
+  posts,
+  error,
+  isFetching,
+  fetchPostsByTagListId,
+  fetchLatestPosts,
+  fetchLatestTags,
+  nPerPage,
+  totalPages,
+  page,
+}) => {
+  useEffect(() => {
+    fetchLatestTags().catch(failAction => {
+      logger.errorReport({
+        report: _.get(failAction, 'payload.error'),
+        message: `Error to fetch latest tags`,
+      })
+    })
+  }, [])
+  useEffect(() => {
+    const catchFunc = failAction => {
+      logger.errorReport({
+        report: _.get(failAction, 'payload.error'),
+        message: `Error to fetch posts (tag id: '${tagId}', page: ${page}, nPerPage: ${nPerPage}).`,
+      })
+    }
+    if (tagId) {
+      fetchPostsByTagListId(tagId, nPerPage, page).catch(catchFunc)
+    } else {
+      fetchLatestPosts(nPerPage, page).catch(catchFunc)
+    }
+  }, [
+    tagId,
+    activeTabIndex,
+    page,
+    nPerPage,
+    fetchPostsByTagListId,
+    fetchLatestPosts,
+  ])
+
+  // Error handling
+  if (error) {
+    return <SystemError error={error} />
+  }
+
+  const title = 'latest' + siteMeta.name.separator + siteMeta.name.full
```

Comment:
'latest' showed up twice(line 299), so we could replace them with constant variable to reduce duplications

Reply:
updated ([ac2670c](https://github.com/twreporter/twreporter-react/pull/2343/commits/ac2670c5c5e4145142df2d4b7a2c6fbbb804121c))
`latest` in line 299 means its redux action list id, thus I export new `LATES_LIST_ID` for it (referr to twreporter-npm-packages [#352](https://github.com/twreporter/twreporter-npm-packages/pull/352))